### PR TITLE
Feature/reform comment

### PR DESCRIPTION
**Pull request by juliusb0205**
@ming-ancino 

Implement reform on user, question create and edit, comments, answers.

### CODE REVIEW

File: app/controllers/answers_controller.rb [hunk 1]
```diff
@@ -2,19 +2,15 @@ class AnswersController < ApplicationController
...
 	def create
 		@question = Question.find params[:question_id]
-		answer = @question.answers.build answerable_params
-		answer.user_id = current_user.id
-		answer.save
+
+		@form = AnswerForm.new current_user.answers.new
+		if @form.validate params[:answer]
+			@form.save
+			@question.answers << @form.model
```
Comment (ming-ancino) [inline app/controllers/answers_controller.rb]: This assignment should be inside the form object.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -9,9 +9,12 @@ def create
...
 		end
 
-		comment = @object.comments.build answerable_params
-		comment.user_id = current_user.id
-		comment.save
+		@form = AnswerForm.new current_user.comments.new
+
+		if @form.validate params[:comment]
+			@form.save
+			@object.comments << @form.model
```
Comment (ming-ancino) [inline app/controllers/comments_controller.rb]: Same here.

File: app/controllers/questions_controller.rb [hunk 3]
```diff
@@ -8,12 +8,12 @@ def index
...
 	def new
-		@question = Question.new
-		@answer = Answer.new
+		@form = QuestionForm.new Question.new
 	end
 
 	def edit
-		@question = Question.find params[:id]
+		@form = QuestionForm.new Question.find params[:id]
+		@form.prepopulate!
```
Comment (ming-ancino) [inline app/controllers/questions_controller.rb]: wtf

File: app/controllers/questions_controller.rb [hunk 4]
```diff
@@ -23,37 +23,34 @@ def show
...
 
 	def create
-		@question = current_user.questions.build question_params
-		if @question.save
-			redirect_to @question
+		@form = QuestionForm.new current_user.questions.new
+
+		if @form.validate params[:question]
+			@form.save
+			redirect_to @form.model
```
Comment (ming-ancino) [inline app/controllers/questions_controller.rb]: Try `@form` only.

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -13,14 +13,16 @@ def show
 	end
 
 	def new
-		@user = User.new
+		new_form
```
Comment (ming-ancino) [inline app/controllers/users_controller.rb]: Research `before_action`

File: app/models/user_form.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+class UserForm < Reform::Form
+  property :name
+  property :email
+  property :password
+  property :password_confirmation
```
Comment (ming-ancino) [inline app/models/user_form.rb]: virtual

File: config/initializers/reform.rb [hunk 7]
```diff
@@ -0,0 +1,4 @@
+require "reform/form/active_model/validations"
```
Comment (ming-ancino) [inline config/initializers/reform.rb]: kick-desk

File: app/models/question_form.rb [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  property :title
+  property :content
+  property :tag_list, :virtual => true,
+     prepopulator: -> (options) { self.tag_list = self.model.tags.map(&:name).join ', ' }
+
+  validates :title, :presence => true
+  validates :content, :presence => true
+
+
+  def save
```
Comment (ming-ancino) [inline app/models/question_form.rb]: what-no

**Comment by juliusb0205** [comment]
@ming-ancino 

Updated forms as requested in comments

File: app/controllers/questions_controller.rb [hunk 9]
```diff
@@ -23,37 +22,36 @@ def show
...
 
 	def update
-		@question = Question.find params[:id]
-		if @question.update_attributes question_params
-			redirect_to @question
+		question = Question.find params[:id]
+		@form = QuestionForm.new question
+
+		if @form.validate params[:question]
+			@form.tags = question.tags
```
Comment (ming-ancino) [inline app/controllers/questions_controller.rb]: ??

**Pull request merged**